cheetah/accelerator/space_charge_kick.py

```diff
@@ -0,0 +1,595 @@
+        new_dims = tuple(dim * 2 for dim in grid_shape)
+
+        # Create a new tensor with the doubled dimensions, filled with zeros
+        new_charge_density = torch.zeros(
+            (self.n_batch,) + new_dims, **self.factory_kwargs
+        )
+
+        # Copy the original charge_density values to the beginning of the new tensor
+        new_charge_density[
+            :,
+            : charge_density.shape[1],
+            : charge_density.shape[2],
+            : charge_density.shape[3],
+        ] = charge_density
+        return new_charge_density
+
+    def _IGF(self, beam: ParticleBeam, cell_size) -> torch.Tensor:
+        """
+        Computes the Integrated Green Function (IGF) with periodic boundary conditions
+        (to perform Hockney's method).
+        """
+        gamma = self._gammaref(beam)
+        dx, dy, ds = (
+            cell_size[:, 0],
+            cell_size[:, 1],
+            cell_size[:, 2] * gamma,
+        )  # scaled by gamma
+        num_grid_points_x, num_grid_points_y, num_grid_points_s = self.grid_shape
+
+        # Create coordinate grids
+        x = torch.arange(num_grid_points_x, **self.factory_kwargs)
+        y = torch.arange(num_grid_points_y, **self.factory_kwargs)
+        s = torch.arange(num_grid_points_s, **self.factory_kwargs)
+        ix_grid, iy_grid, is_grid = torch.meshgrid(x, y, s, indexing="ij")
+        x_grid = (
+            ix_grid[None, :, :, :] * dx[:, None, None, None]
+        )  # Shape: [n_batch, nx, ny, nz]
+        y_grid = (
+            iy_grid[None, :, :, :] * dy[:, None, None, None]
+        )  # Shape: [n_batch, nx, ny, nz]
+        s_grid = (
+            is_grid[None, :, :, :] * ds[:, None, None, None]
+        )  # Shape: [n_batch, nx, ny, nz]
+
+        # Compute the Green's function values
+        G_values = (
+            self._integrated_potential(
+                x_grid + 0.5 * dx, y_grid + 0.5 * dy, s_grid + 0.5 * ds
+            )
+            - self._integrated_potential(
+                x_grid - 0.5 * dx, y_grid + 0.5 * dy, s_grid + 0.5 * ds
+            )
+            - self._integrated_potential(
+                x_grid + 0.5 * dx, y_grid - 0.5 * dy, s_grid + 0.5 * ds
+            )
+            - self._integrated_potential(
+                x_grid + 0.5 * dx, y_grid + 0.5 * dy, s_grid - 0.5 * ds
+            )
+            + self._integrated_potential(
+                x_grid + 0.5 * dx, y_grid - 0.5 * dy, s_grid - 0.5 * ds
+            )
+            + self._integrated_potential(
+                x_grid - 0.5 * dx, y_grid + 0.5 * dy, s_grid - 0.5 * ds
+            )
+            + self._integrated_potential(
+                x_grid - 0.5 * dx, y_grid - 0.5 * dy, s_grid + 0.5 * ds
+            )
+            - self._integrated_potential(
+                x_grid - 0.5 * dx, y_grid - 0.5 * dy, s_grid - 0.5 * ds
+            )
+        )
+
+        # Initialize the grid with double dimensions
+        green_func = torch.zeros(
```

Comment:
`green_func` is probably also a bit confusing here because it's not a function, but whatever that function return, right?

Reply:
Agreed. How about `green_func_values`?